### ISSUE TITLE
No training executed when resume training using diffrent training dataset than initially used for pre-train

### ISSUE THREAD

**Issue by balandongiv** [open-mmlab/mmocr#1111]
I manage to train `crnn` using the following cfg

```
_base_ = [
    'configs/_base_/default_runtime.py',
    'configs/_base_/recog_models/crnn.py',
    'configs/_base_/recog_pipelines/crnn_pipeline.py',
    'configs/_base_/schedules/schedule_adadelta_5e.py'
]
```
with the `Syn90k` as training dataset, and  `IIIT5K` & `SVT` as validation set. The metric at `6` epoch is as follow

`"0_word_acc_ignore_case": 0.824, "0_word_acc_ignore_case_symbol": 0.824, "0_1-N.E.D": 0.9334, "1_char_recall": 0.9297, "1_char_precision": 0.9569, "1_word_acc": 0.8099, "1_word_acc_ignore_case": 0.8099, "1_word_acc_ignore_case_symbol": 0.8099, "1_1-N.E.D": 0.9224, "mean_char_recall": 0.9395, "mean_char_precision": 0.95515, "mean_word_acc": 0.43395, "mean_word_acc_ignore_case": 0.81695, "mean_word_acc_ignore_case_symbol": 0.81695, "mean_1-N.E.D": 0.9279}`

Given the overall good result, I intend to use the ` epoch_6.pth` as a base for `transfer learning` on different `training` and `validation` dataset.

Hence the following `cfg`

```
_base_ = [
    '/mmocr/configs/_base_/default_runtime.py',
    '/mmocr/configs/_base_/recog_models/crnn.py',
    '/mmocr/configs/_base_/recog_pipelines/crnn_pipeline.py',
    '/mmocr/configs/_base_/schedules/schedule_adadelta_5e.py'
]

load_from ='/content/mmocr_debug/epoch_6.pth'



# Set up working dir to save files and logs.
work_dir ='/content/mmocr_debug/resume'


train_img_prefix ='/content/mmocr_debug/train/mmocr_compatible_annotation/crops'
train_ann_filex = '/content/mmocr_debug/train/mmocr_compatible_annotation/train_label.jsonl'


train_img_prefix ='/content/mmocr_debug/val/mmocr_compatible_annotation/crops'
train_ann_filex = '/content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl'


train1 = dict(type='OCRDataset',
              img_prefix=train_img_prefix,
              ann_file=train_ann_filex,
              loader=dict(type='AnnFileLoader',
                          repeat=1,
                          file_format='txt',
                          file_storage_backend='disk',
                          parser=dict(type='LineJsonParser',
                                      keys=['filename', 'text'])),
              pipeline=None,
              test_mode=False)




test1 = dict(type='OCRDataset',
             img_prefix=train_img_prefix,
             ann_file=train_ann_filex,
             loader=dict(type='AnnFileLoader',
                         repeat=1,
                         file_format='txt',  # only txt and lmdb
                         file_storage_backend='disk',
                         parser=dict(type='LineJsonParser',
                                     keys=['filename', 'text'])),
             pipeline=None,           # why the pipeline is set to None?
             test_mode=False)



test_list = [test1]
train_list=[train1]
#####

train_pipeline = {{_base_.train_pipeline}}
test_pipeline = {{_base_.test_pipeline}}

data = dict(
    samples_per_gpu=64,
    workers_per_gpu=4,
    val_dataloader=dict(samples_per_gpu=1),
    test_dataloader=dict(samples_per_gpu=1),
    train=dict(
        type='UniformConcatDataset',
        datasets=train_list,
        pipeline=train_pipeline),
    val=dict(
        type='UniformConcatDataset',
        datasets=test_list,
        pipeline=test_pipeline),
    test=dict(
        type='UniformConcatDataset',
        datasets=test_list,
        pipeline=test_pipeline))


evaluation = dict(interval=1, metric='acc')

cudnn_benchmark = True
```

However, upon executing the `train_detector(model, datasets, cfg, distributed=False, validate=True)`, I notice there is no `training` been conducted at the background. I got this notion upon visually check the following output.

```
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 84.2 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:18,271 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:18,274 - mmocr - INFO - Epoch(val) [1][63]	0_char_recall: 0.7937, 0_char_precision: 0.8130, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.6190, 0_word_acc_ignore_case_symbol: 0.6190, 0_1-N.E.D: 0.7937
2022-06-24 12:43:20,605 - mmocr - INFO - Saving checkpoint at 2 epochs

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 85.4 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:21,813 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:21,819 - mmocr - INFO - Epoch(val) [2][63]	0_char_recall: 0.8175, 0_char_precision: 0.8240, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.6508, 0_word_acc_ignore_case_symbol: 0.6508, 0_1-N.E.D: 0.8175

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 90.9 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:24,948 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:24,955 - mmocr - INFO - Epoch(val) [3][63]	0_char_recall: 0.8095, 0_char_precision: 0.8160, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.6349, 0_word_acc_ignore_case_symbol: 0.6349, 0_1-N.E.D: 0.8095
2022-06-24 12:43:27,273 - mmocr - INFO - Saving checkpoint at 4 epochs

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 83.3 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:28,577 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:28,584 - mmocr - INFO - Epoch(val) [4][63]	0_char_recall: 0.8571, 0_char_precision: 0.8640, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.7302, 0_word_acc_ignore_case_symbol: 0.7302, 0_1-N.E.D: 0.8571

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 85.8 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:31,775 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:31,781 - mmocr - INFO - Epoch(val) [5][63]	0_char_recall: 0.8651, 0_char_precision: 0.8651, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.7460, 0_word_acc_ignore_case_symbol: 0.7460, 0_1-N.E.D: 0.8651
2022-06-24 12:43:34,117 - mmocr - INFO - Saving checkpoint at 6 epochs

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 82.1 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:35,426 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:35,430 - mmocr - INFO - Epoch(val) [6][63]	0_char_recall: 0.8810, 0_char_precision: 0.8810, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.7778, 0_word_acc_ignore_case_symbol: 0.7778, 0_1-N.E.D: 0.8810

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 87.5 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:38,587 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:38,592 - mmocr - INFO - Epoch(val) [7][63]	0_char_recall: 0.8810, 0_char_precision: 0.8810, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.7778, 0_word_acc_ignore_case_symbol: 0.7778, 0_1-N.E.D: 0.8810
2022-06-24 12:43:40,911 - mmocr - INFO - Saving checkpoint at 8 epochs

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 88.4 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:42,096 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:42,102 - mmocr - INFO - Epoch(val) [8][63]	0_char_recall: 0.8810, 0_char_precision: 0.8810, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.7778, 0_word_acc_ignore_case_symbol: 0.7778, 0_1-N.E.D: 0.8810

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 85.8 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:45,276 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:45,283 - mmocr - INFO - Epoch(val) [9][63]	0_char_recall: 0.8968, 0_char_precision: 0.8968, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.8095, 0_word_acc_ignore_case_symbol: 0.8095, 0_1-N.E.D: 0.8968
2022-06-24 12:43:47,620 - mmocr - INFO - Saving checkpoint at 10 epochs

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 83.8 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:48,887 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:48,894 - mmocr - INFO - Epoch(val) [10][63]	0_char_recall: 0.8889, 0_char_precision: 0.8960, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.7937, 0_word_acc_ignore_case_symbol: 0.7937, 0_1-N.E.D: 0.8889

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 86.3 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:52,073 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:52,079 - mmocr - INFO - Epoch(val) [11][63]	0_char_recall: 0.8968, 0_char_precision: 0.8968, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.8095, 0_word_acc_ignore_case_symbol: 0.8095, 0_1-N.E.D: 0.8968
2022-06-24 12:43:54,523 - mmocr - INFO - Saving checkpoint at 12 epochs

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 62.5 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:56,127 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:56,130 - mmocr - INFO - Epoch(val) [12][63]	0_char_recall: 0.9048, 0_char_precision: 0.9048, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.8254, 0_word_acc_ignore_case_symbol: 0.8254, 0_1-N.E.D: 0.9048

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 86.9 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:43:59,299 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:43:59,304 - mmocr - INFO - Epoch(val) [13][63]	0_char_recall: 0.9048, 0_char_precision: 0.9048, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.8254, 0_word_acc_ignore_case_symbol: 0.8254, 0_1-N.E.D: 0.9048
2022-06-24 12:44:01,639 - mmocr - INFO - Saving checkpoint at 14 epochs

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 63/63, 86.9 task/s, elapsed: 1s, ETA:     0s

2022-06-24 12:44:02,833 - mmocr - INFO - 
Evaluating /content/mmocr_debug/val/mmocr_compatible_annotation/train_label.jsonl with 63 images now
2022-06-24 12:44:02,839 - mmocr - INFO - Epoch(val) [14][63]	0_char_recall: 0.9127, 0_char_precision: 0.9127, 0_word_acc: 0.0000, 0_word_acc_ignore_case: 0.8413, 0_word_acc_ignore_case_symbol: 0.8413, 0_1-N.E.D: 0.9127


```

May I know what did I do wrong? The issue can be reproduced via the [Notebook](https://github.com/balandongiv/mmocr_tutorial/blob/main/DEBUG_UNABLE_TO_RESUME_TRAINING.ipynb)

Update.
I just realize whereby train a model from scratch also produce some issue as shown in this [mmocr_tutorial_custom](https://github.com/balandongiv/mmocr_tutorial/blob/main/custom%20mmocr.ipynb)

**Comment by balandongiv** [comment]
Great catch @Mountchicken , really appreciate it!
